src/hotspot/share/oops/markWord.cpp

```diff
@@ -98,3 +99,27 @@ void markWord::print_on(outputStream* st, bool print_monitor_info) const {
     st->print(" age=%d)", age());
   }
 }
+
+#ifdef _LP64
+narrowKlass markWord::narrow_klass() const {
+  return narrowKlass(value() >> klass_shift);
+}
+
+Klass* markWord::klass() const {
+  return CompressedKlassPointers::decode(narrow_klass());
+}
+
+markWord markWord::set_narrow_klass(const narrowKlass nklass) const {
+  return markWord((value() & ~klass_mask_in_place) | ((uintptr_t) nklass << klass_shift));
+}
+
+markWord markWord::set_klass(const Klass* klass) const {
+  if (UseCompressedClassPointers) {
+    // TODO: Don't cast to non-const, change CKP::encode() to accept const Klass* instead.
+    narrowKlass nklass = CompressedKlassPointers::encode(const_cast<Klass*>(klass));
+    return set_narrow_klass(nklass);
+  } else {
+    return markWord(value());
```

Comment:
So this `else` path does not use `klass`? Is that because `UseCompressedClassPointers` is always `true`? Is it, though, on x86_32? It would be cleaner to put `ShouldNotReachHere()` if that path should not be taken.

Reply:
In 32bit, we should not call this (the whole block is _LP64). In 64bit, UseCompressedClassPointers is forced and the else-branch should not be used/eventually go away. Yes, we can do ShouldNotReachHere() there.